src/sentry/api/endpoints/organization_member_invite/index.py

```diff
@@ -0,0 +1,337 @@
+                return Response(
+                    {"detail": "You cannot assign members to teams you are not a member of."},
+                    status=400,
+                )
+
+        if has_teams and not organization_roles.get(assigned_org_role).is_team_roles_allowed:
+            return Response(
+                {
+                    "email": f"The user with a '{assigned_org_role}' role cannot have team-level permissions."
+                },
+                status=400,
+            )
+
+        with transaction.atomic(router.db_for_write(OrganizationMemberInvite)):
+            teams = []
+            for team in result.get("teams", []):
+                teams.append({"id": team.id, "slug": team.slug, "role": None})
+
+            omi = OrganizationMemberInvite(
+                organization=organization,
+                email=result["email"],
+                role=result["orgRole"],
+                inviter_id=request.user.id,
+                organization_member_team_data=teams,
+            )
+
+            omi.save()
+
+        if result.get("sendInvite"):
+            referrer = request.query_params.get("referrer")
+            omi.send_invite_email(referrer)
+            member_invited.send_robust(
+                member=omi, user=request.user, sender=self, referrer=request.data.get("referrer")
+            )
+
+        self.create_audit_entry(
+            request=request,
+            organization_id=organization.id,
+            target_object=omi.id,
+            data=omi.get_audit_log_data(),
+            event=(audit_log.get_event_id("MEMBER_INVITE")),
+        )
+
+        return Response(serialize(omi), status=201)
+
+    def _request_to_invite_member(self, request: Request, organization) -> Response:
+        serializer = OrganizationMemberInviteRequestSerializer(
+            data=request.data,
+            context={"organization": organization, "allowed_roles": roles.get_all()},
+        )
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+        result = serializer.validated_data
+
+        with transaction.atomic(router.db_for_write(OrganizationMemberInvite)):
+            teams = []
+            for team in result.get("teams", []):
+                teams.append({"id": team.id, "slug": team.slug, "role": None})
+
+            omi = OrganizationMemberInvite(
+                organization=organization,
+                email=result["email"],
+                role=result["orgRole"],
+                inviter_id=request.user.id,
+                organization_member_team_data=teams,
+                invite_status=InviteStatus.REQUESTED_TO_BE_INVITED.value,
+            )
+
+            omi.save()
+
+        self.create_audit_entry(
+            request=request,
+            organization_id=organization.id,
+            target_object=omi.id,
+            data=omi.get_audit_log_data(),
+            event=(audit_log.get_event_id("INVITE_REQUEST_ADD")),
+        )
```

Comment:
this logic is the same as adding an approved invite save for `invite_status` and `event`, we could use a single function for both